Cloud Freemium notice to self-hosted admins & users

jasonblais opened the pull request:
What the notice will look like. Text copy reviewed by Ashley Dotterweich, design prepared by Anneliese Klein.

<img width="437" alt="image" src="https://user-images.githubusercontent.com/13119842/173615161-f8288dc3-c723-4241-a4ca-e199cf14a599.png">

esethna commented on notices.json:
```diff
@@ -1,4 +1,24 @@
+      "clientType": "all",
+      "sku": "e0",
+      "instanceType": "onprem",
+      "repeatable": true,
```
This flag means all our on-prem users will see it everyday for three days, so it may be overly disruptive.

I'm also not sure how well the repeatable flag has been tested. Could I propose that we don't use that flag, and if we want to refire the notice sometime in the future we just submit a new one?

jasonblais replied:
:+1: sounds good, let's do that. Removed the flag

esethna commented on notices.json:
```diff
@@ -1,4 +1,23 @@
+    "conditions": {
+      "audience": "all",
+      "clientType": "all",
+      "sku": "e0",
```
@ogi-m please make sure this flag is thoroughly tested since we've changed our packaging since notices were built (hopefully the SKU names didn't change on the back-end though)

ogi-m commented:
@esethna @jasonblais Just to confirm - does the `e0` sku mean that only enterprise installations without a license should be getting this notice?

jasonblais commented:
> does the e0 sku mean that only enterprise installations without a license should be getting this notice?

That's correct

jasonblais commented:
FYI, made two changes
 - changed date to June 21 onward (to avoid weekend effect if merged before Monday)
 - added `utm_source` tag to the link: `https://mattermost.com/sign-up/?utm_source=freemium-inapp-notice`

ogi-m reviewed (changes requested):
@jasonblais The image link appears to be invalid

jasonblais commented:
@ogi-m it's because the image hasn't been committed to the `release` branch yet, it's included in this PR.

The image renders fine on the `jasonblais-patch-1` branch: https://raw.githubusercontent.com/mattermost/notices/jasonblais-patch-1/images/mattermost-cloud-freemium.png

ogi-m commented:
Ah ok thanks @jasonblais. This is good to merge then - verified that the notice shows up for a server on e0, and doesn't show up on servers running e10/e20
